curriculum/outlines/Linear DS.md

```diff
@@ -0,0 +1,39 @@
+	 3. 3D Static Array
+2. Dynamically-Resizeable Array
+3. BitSet
+4. Bitmasks (Basic bit manipulation)
+	  * Example: Flipping i-th bit in an integer.
+5. LinkedLists
+6. Stacks
+7. Queues
+8. Dequeues
+    
+---
+## Problem Sets
+
+### Problem Set #1
+
+| Problem        | Tags          | Notes  | Solution |
+|:------------- |:-------------|:-----|:--------|
+| [Boy or Girl](http://codeforces.com/problemset/problem/236/A)      |  1D Array manipulation   | use a boolean array to keep track whether a character is found or not.    | [Link](http://codeforces.com/contest/236/submission/33829251) |
+| [Splitting Numbers](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=9)      |  Bit Manipulation   | basic bit manipulation but could be solved with strings + binary decimal conversion.   | [Link](https://ideone.com/VcuwoQ) |
+| [Free Spots](https://uva.onlinejudge.org/index.php?option=onlinejudge&page=show_problem&problem=1644)      |  2D Array Manipulation    | use 2D boolean array of size 500 × 500)   | [Link](https://github.com/fernandohbc/MyPCS/blob/master/uvaonlinejudge/src/volume_cvii/P10703_FreeSpots.java) |
+| [Error Correction](https://uva.onlinejudge.org/index.php?option=onlinejudge&page=show_problem&problem=482)      |  2D Array Manipulation    | count the number of '1's for each row/col, all of them must be even , if there is an error then check if it is on the same row and col   | [Link](https://github.com/izharishaksa/UVa-Solution/blob/master/src/datastructures/builtin/Problem541YES.java) |
+| [Rails](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=455)      |  Stacks    | use a stack to simulate the process   | [Link](https://ideone.com/qetrPf) |
+
+### Problem Set #2
+
+| Problem        | Tags          | Notes  | Solution |
+|:------------- |:-------------|:-----|:--------|
+| [Throwing cards away I](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=1876)      | Queues    | simulate the process using a queue   | [Link](https://ideone.com/RbaMDN) |
+| [Broken Keyboard](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=3139)      |  Dequeues , Linked lists    | think of a data structure that allow insertion from both sides   | [Link](https://ideone.com/tuKwEl) |
+| [Machined Surfaces](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=355)      |  1D Array Manipulation    | count the number of 'X's in each row + keep track of their max   | [Link](https://ideone.com/VK956q) |
+| [Transform the Expression](http://www.spoj.com/problems/ONP/)      |  Stacks    | tricky way to use a stack , not direct   | [Link](https://ideone.com/urywUI) |
+| [The Most Potent Corner](https://uva.onlinejudge.org/index.php?option=com_onlinejudge&Itemid=8&page=show_problem&problem=1205)      | Bitmasks    | _   | [Link](https://ideone.com/QTrOoD) |
+
```

Comment:
Add problem set 3

Reply:
We added just two problem sets because Linear DS was assigned just 1 ACM day, and we agreed in the meeting that each ACM day will contain 10 problems, right ?

Reply:
10 Problems per ACM day means they will get two problem sets the first one is topic-oriented which is problem set 1. The second one is a general problem set that contains problems from topics they already know. Other problem sets (2 and 3) are for people who want to practice more on this topic

Reply:
This implies that problem set 1 problems should be selected such that they are enough for people who will not practice more than one ACM day.

Reply:
We had 8 types of linear data structures so we had to put at least 8 topic-oriented problems (2 problem sets), and we will add problem set 3 which will contain 5 general problems (for sure on topics they already know), ok ?

Reply:
general problems are not added in the outline. They will be added in the main plan because they are irrelevant to a specific topic outline